core/foundation/src/RConversionRuleParser.cxx

```diff
@@ -336,12 +335,14 @@ namespace ROOT
             error_string += warning + " - the list of checksums is empty\n";
          }
 
-         for( lsIt = lst.begin(); lsIt != lst.end(); ++lsIt )
-            if( !TSchemaRuleProcessor::IsANumber( *lsIt ) ) {
-               error_string = warning + " - " + *lsIt + " is not a valid value";
-               error_string += " of checksum parameter - an integer expected";
+         for( const auto& chk : lst ) {
+            if( !TSchemaRuleProcessor::IsANumber(chk)
```

Comment:
Since it does not matter (at least not used yet) that a number is an hex, wouldn't it make sense to just make IsANumber return true for hex?

Reply:
That can be done, but it's not too easy, as other functions also use it (for e.g. checking that the version is a number). These would be thrown off when you give the version as hex.

Reply:
Ok, a simple solution is to add a flag that allows to additionally accept hex.